ext/drutil/drutil.c

```diff
@@ -71,23 +71,31 @@ static int drutil_init_count;
 #ifdef X86
 
 static inline void
-native_unix_cpuid(uint *eax, uint *ebx, uint *ecx, uint *edx)
+native_unix_cpuid(uint *xax, uint *xbx, uint *xcx, uint *xdx)
 {
 #    ifdef UNIX
-    /* We need to do this ebx trick, because ebx might be used for fPIC,
+    /* We need to do this xbx trick, because xbx might be used for fPIC,
      * and gcc < 5 chokes on it. This can get removed and replaced by
      * a "=b" constraint when moving to gcc-5.
      */
-    asm volatile("xchgl\t%%ebx, %k1\n\t"
+#        ifdef X64
```

Comment:
In general our precedence is to use ASM_XAX to avoid needing a full ifdef and duplicating the while sequence: it's defined for the core and tests, not sure if it's easy to get those same defines over here via the header used for tests.

Reply:
Also need xchgl vs. xchgq, and it needs the %q and %k constraints to constrain the register into si or dimode.